### PR TITLE
Direct logout

### LINKED ISSUE #2072: Remove extra logout screen
### Your use case

Go to setting and tap "log-out".
- The app is opening an extra screen.
- The user has to tap "Log-Out" again.
- A pop-up is shown: "are you sure you want to logout?"

Expectation:

- Do not show the extra screen
- Show the pop-up right away (same as iOS)

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : simplify flow

## Content

In the case of not last session and all keys are backed up, do not go to the logout flow, but just render a confirmation dialog.

I do not want to remove the existing screen, which should not be used anymore for several reasons:
- the state is live so this screen may be useful
- I expect at some point that we improve the existing screen to notifiy user that this is maybe not necessary to log out, as per other messaging app. And so this code could be removed.

## Motivation and context

Closes #2072 

## Tests

- Login to a session where other sessions exist
- Log out and observe that there is only a confirmation dialog which is displayed.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/root/PreferencesRootView.kt [hunk 1]
```diff
@@ -162,7 +163,13 @@ fun PreferencesRootView(
             headlineContent = { Text(stringResource(id = CommonStrings.action_signout)) },
             leadingContent = ListItemContent.Icon(IconSource.Resource(CommonDrawables.ic_sign_out)),
             style = ListItemStyle.Destructive,
-            onClick = onSignOutClicked,
+            onClick = {
+                if (state.directLogoutState.canDoDirectSignOut) {
+                    state.directLogoutState.eventSink(DirectLogoutEvents.Logout(ignoreSdkError = false))
```
Comment (jmartinesp) [inline features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/root/PreferencesRootView.kt:168]: Maybe we could change the `onSignOutClicked` lambda so it takes a `direct: Boolean` param and handle the logic there?

**Review by jmartinesp — APPROVED** [review]
Thanks for the changes! I have a couple of suggestions, but feel free to ignore them if they make things too complicated.

Also, there was an unexpected change in screenshots if I'm not wrong. Although it might not be caused by this PR.

File: features/logout/impl/src/main/kotlin/io/element/android/features/logout/impl/tools/Extensions.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package io.element.android.features.logout.impl.tools
+
+import io.element.android.libraries.matrix.api.encryption.BackupUploadState
+import io.element.android.libraries.matrix.api.encryption.SteadyStateException
+
+internal fun BackupUploadState.isBackingUp(): Boolean {
+    return when (this) {
+        BackupUploadState.Waiting,
+        is BackupUploadState.Uploading -> true
+        is BackupUploadState.SteadyException -> exception is SteadyStateException.Connection
```
[the remark this answers is not in the document]
Comment (bmarty) [reply features/logout/impl/src/main/kotlin/io/element/android/features/logout/impl/tools/Extensions.kt:26]: Yes

File: features/logout/impl/src/main/kotlin/io/element/android/features/logout/impl/ui/LogoutActionDialog.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    state: Async<String?>,
+    onForceLogoutClicked: () -> Unit,
+    onDismissError: () -> Unit,
+    onSuccessLogout: (String?) -> Unit,
+) {
+    when (state) {
+        is Async.Loading ->
+            ProgressDialog(text = stringResource(id = R.string.screen_signout_in_progress_dialog_content))
+        is Async.Failure ->
+            ConfirmationDialog(
```
[the remark this answers is not in the document]
Comment (bmarty) [reply features/logout/impl/src/main/kotlin/io/element/android/features/logout/impl/ui/LogoutActionDialog.kt:39]: I will change it, hence the retry will not do exactly the same thing, i.e. any error from the sdk/server will be ignored.

File: features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/root/PreferencesRootView.kt (hunk 1 above)
Comment (bmarty) [reply features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/root/PreferencesRootView.kt:168]: Moved to the Node, which also knows the state, so no need for a parameter.

**Pull request merged**